### PR TITLE
Rework mod content installation.

### PR DESCRIPTION
**Pull request by pchote**
This PR removes the magic cross-mod inheritance that was originally implemented to support the long-defunct oramod archive format.

The key benefits here are:
 * Removal of implicit and magic behaviours around mod.yaml, filesystems, and translations
 * Clean separation between installation and runtime (mods that ship all their own assets can now completely ignore the installation logic 🎉)
 * Another step towards upstreaming TDHD by allowing first-class support for mod-specific installation behaviours

### CODE REVIEW

File: mods/ra-content/mod.yaml [hunk 1]
```diff
@@ -0,0 +1,139 @@
+Metadata:
+	Title: mod-title
+	Version: {DEV_VERSION}
+	Hidden: true
+
+FileSystem: DefaultFileSystem
+    Packages:
```
Comment (PunkPun) [inline mods/ra-content/mod.yaml]: spaces instead of tabs, this is the case in most mod.yamls

**Review by PunkPun — COMMENTED** [review]
should this supersede #21609 and fully embrace new localisation instead of only partially?

Just run ./utility.sh cnc-content --extract-chrome-strings and deduplicate a few that already exist in common en.ftl like

button-cancel = Cancel
button-retry = Retry
button-back = Back
button-continue = Continue
button-quit = Quit

**Comment by pchote** [comment]
> should this supersede https://github.com/OpenRA/OpenRA/pull/21609 and fully embrace new localisation instead of only partially?

Can do if you'd like, although I was expecting that one would be merged and this one rebased.

**Comment by PunkPun** [comment]
you can merge it now and rebase then

**Comment by PunkPun** [comment]
installation from remaster seems to be crashing

System.InvalidOperationException: Cannot locate type: ExtractMixSourceAction
   at OpenRA.ObjectCreator.CreateObject[T](String className, Dictionary`2 args) in /OpenRA/OpenRA.Game/ObjectCreator.cs:line 109
   at OpenRA.Mods.Common.Widgets.Logic.InstallFromSourceLogic.<>c__DisplayClass44_0.<InstallFromSource>g__RunSourceActions|2(MiniYamlNode contentPackageYaml, <>c__DisplayClass44_1& ) in /OpenRA/OpenRA.Mods.Common/Widgets/Logic/Installation/InstallFromSourceLogic.cs:line 261

**Comment by pchote** [comment]
Fixed and rebased.

**Review by PunkPun — COMMENTED** [review]
- D2K detection of assets is wrong! While the quick install itself doesn't fail, the game fails to load and puts you back into the d2k-content mod. Note that with the installed files you are able to play on bleed, just not on this branch
- after I install everything from remastered it is no longer detected
- after installing required files from the remaster, I still have the option to quick install (no warning)

**Comment by pchote** [comment]
Rebased, fixed, and tested quick install + at least one advanced install method for all mods.

> D2K detection of assets is wrong! 

Oops, thats what I get for rushing things without proper testing.

> after I install everything from remastered it is no longer detected

If you mean for RA, then this is expected because the remaster doesn't include the soviet briefings and therefore isn't a valid source for that content package.

> after installing required files from the remaster, I still have the option to quick install (no warning)

This actually has always been the case, so i'd prefer to leave any changes to this to a different PR!

**Review by PunkPun — APPROVED** [review]
LGTM, we'll want more thorough playtesting when playtest launch

**Comment by PunkPun** [comment]
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/11601f5989a514117cfbfe7f5e671cceaa76778f...ed9f26156517e048a5c8d3f722e88cbea01953b1)

**Pull request merged**